### PR TITLE
fix: Remove deprecated parameter `vsync` from `AnimatedSize`

### PR DESCRIPTION
**Pull request by xfqwdsj**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description

Closes: #861

### Approach

Remove the specified `vsync` property.

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry

### CODE REVIEW

**Review by Nidal-Bakir — APPROVED** [review]

**Review by Nidal-Bakir — CHANGES_REQUESTED** [review]
Could you please add a changelog entry and bump the version?

**Comment by xfqwdsj** [comment]
> Could you please add a changelog entry and bump the version?

OK  I am trying.  Thank you!

**Review by Nidal-Bakir — CHANGES_REQUESTED** [review]
You also can remove this line.

`with SingleTickerProviderStateMixin`

**Comment by xfqwdsj** [comment]
> You also can remove this line.
> 
> `with SingleTickerProviderStateMixin`

finished! Please check it out, thank you.

File: packages/flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## Unreleased (master branch)
+
+### Bug Fixes
+
+* Since Flutter 3.8, the `vsync` property was removed from `AnimatedSize` causing compilation failure. ([#861](https://github.com/parse-community/Parse-SDK-Flutter/issues/861))
```
Comment (Nidal-Bakir) [inline packages/flutter/CHANGELOG.md:5]:
```
[4.0.1](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-4.0.0...flutter-4.0.1) (2023-03-25)

### Bug Fixes

* Remove deprecated parameter `vsync` form`AnimatedSize` ([#864](https://github.com/parse-community/Parse-SDK-Flutter/pull/864))

```

Comment (xfqwdsj) [reply packages/flutter/CHANGELOG.md]: completed.

**Review by Nidal-Bakir — DISMISSED** [review]
Thanks, Looks good.

**Comment by mtrezza** [comment]
If we remove a property and therefore remove support for currently supported Flutter SDK versions then this is a breaking change. Let's discuss this [in the issue](https://github.com/parse-community/Parse-SDK-Flutter/issues/861#issuecomment-1483899429).

File: packages/flutter/lib/src/utils/parse_live_list.dart [hunk 2]
```diff
@@ -272,8 +271,6 @@ class _ParseLiveListElementWidgetState<T extends sdk.ParseObject>
       sizeFactor: widget.sizeFactor,
       child: AnimatedSize(
         duration: widget.duration,
-        // ignore: deprecated_member_use
-        vsync: this,
```
Comment (mbfakourii) [inline packages/flutter/lib/src/utils/parse_live_list.dart:276]: Is there no alternative introduced for vsync?

**Review by mbfakourii — CHANGES_REQUESTED** [review]

File: packages/flutter/lib/src/utils/parse_live_list.dart (hunk 2 above)
Comment (xfqwdsj) [reply packages/flutter/lib/src/utils/parse_live_list.dart:276]: According to [Flutter's commit](packages/flutter/lib/src/widgets/animated_size.dart), this property is marked as deprecated and its value is ignored.  So there is no need to introduce other alternatives.  Thanks!

Comment (mbfakourii) [reply packages/flutter/lib/src/utils/parse_live_list.dart:276]: I get it, thank you

**Review by mbfakourii — APPROVED** [review]

**Comment by mbfakourii** [comment]
> If we remove a property and therefore remove support for currently supported Flutter SDK versions then this is a breaking change. Let's discuss this [in the issue](https://github.com/parse-community/Parse-SDK-Flutter/issues/861#issuecomment-1483899429).

I don't think it is a breaking change. Because it is a deprecated variable. Also, this issue and change is inside Parse and does not affect the api

**Comment by mtrezza** [comment]
@mbfakourii Could you comment in the issue (see link in my previous comment)? From the discussion there it seems that this is in fact a breaking change.

**Comment by mbfakourii** [comment]
> @mbfakourii Could you comment in the issue (see link in my previous comment)? From the discussion there it seems that this is in fact a breaking change.

I checked, is it true that I think it is a breaking change, with these qualities, we have a lot of breaking change in PRs!?

**Comment by Nidal-Bakir** [comment]
@mtrezza 
we can start with this PR

**Comment by mtrezza** [comment]
Is this PR a breaking change?

**Comment by Nidal-Bakir** [comment]
No, its not

**Comment by mtrezza** [comment]
Sorry, I forgot that there was discussion here already about this. So since removing a deprecated param is breaking, I've added an additional entry in the BREAKING CHANGES section.

**Comment by Nidal-Bakir** [comment]
The issue with CI is related to this PR and issue #877. Once we merge/fix them, the CI tests for Flutter should pass.

**Comment by xfqwdsj** [comment]
This is great!  Thanks for your work!

**Pull request merged**